src/manifolds/SegreWarped.jl

```diff
@@ -0,0 +1,284 @@
+        \dots,\\
+        x_d \cos\mathopen{\Big(} \frac{g \lVert u_d \rVert_{x_d}}{A m} \mathclose{\Big)} + \frac{u_d}{\lVert u_d \rVert_{x_d}} \sin\mathopen{\Big(} \frac{g \lVert u_d \rVert_{x_d}}{A m} \mathclose{\Big)}
+    \right),
+````
+where
+````math
+    g = \tan^{-1}\mathopen{\Big(} t \frac{\sqrt{P^2 + 1}}{\lambda} + P \mathclose{\Big)} - \tan^{-1}(P),\\
+    m = \sqrt{\lVert u_1 \rVert_{x_1}^2 + \dots + \lVert u_d \rVert_{x_d}^2},\\
+    P = \frac{\nu}{\lambda A m},\\
+    t = \lVert v \rVert_{p}.
+````
+If ``m = 0`` and ``\nu t < \lambda``, then ``\operatorname{exp}_p(v) = p + v``.
+
+For a proof, see proposition 3.1 in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+"""
+exp(M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}}, p, v) where {V,A}
+
+function exp!(M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}}, q, p, v) where {V,A}
+    m_ = sqrt(
+        sum([
+            norm(Sphere(n - 1), x, xdot)^2 for (n, x, xdot) in zip(V, p[2:end], v[2:end])
+        ]),
+    )
+
+    q[1][1] = sqrt((p[1][1] + v[1][1])^2 + (p[1][1] * A * m_)^2)
+
+    f = pi / 2 - atan((p[1][1] + v[1][1]) / (p[1][1] * A * m_))
+    if m_ == 0
+        for (x, y) in zip(p[2:end], q[2:end])
+            y .= x
+        end
+    else
+        for (n, x, y, xdot) in zip(V, p[2:end], q[2:end], v[2:end])
+            a = norm(Sphere(n - 1), x, xdot)
+            y .=
+                x * cos(a * f / (A * m_)) .+
+                xdot * (f / (A * m_)) * sinc(a * f / (A * m_ * pi))
+        end
+    end
+
+    return 0
+end
+
+@doc raw"""
+    function log(M::MetricManifold{ℝ, Segre{ℝ,V}, WarpedMetric{A}}, p, q)
+
+Logarithmic map on the warped Segre manifold.
+
+Let ``p \doteq (\lambda, x_1, \dots, x_d)``, ``q \doteq (\mu, y_1, \dots, y_d) \in \mathcal{S}_A``.
+Assume ``p`` and ``q`` are connected by a geodesic.
+Let
+````math
+    m = \sqrt{\sphericalangle(x_1, y_1)^2 + \dots + \sphericalangle(x_d, y_d)^2}
+````
+and assume ``(\mu, y_1, \dots, y_d)`` is the representation of ``q`` that minimizes ``m``. Then
+````math
+    \operatorname{log}_p(q) =
+    c \left(
+        \frac{\lambda A m \mathopen{\Big(} \cos(A m) - \frac{\lambda}{\mu} \mathclose{\Big)}}{\sin(A m)},
+        \frac{\sphericalangle(x_1, y_1) (y_1 - \langle x_1, y_1 \rangle x_1)}{\sin(\sphericalangle(x_1, y_1))},
+        \dots,
+        \frac{\sphericalangle(x_d, y_d) (y_d - \langle x_d, y_d \rangle x_d)}{\sin(\sphericalangle(x_d, y_d))}
+    \right),
+````
+where ``c`` is determined by ``\lVert \operatorname{log}_p(q) \rVert_{p} = \operatorname{dist}(p, q)``.
+
+For a proof, see theorem 4.4 in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+"""
+function log(M::MetricManifold{ℝ,Segre{ℝ,V},WarpedMetric{A}}, p, q) where {V,A}
+    q_ = closest_representation(M, p, q)
+    if connected_by_geodesic(M, p, q)
+        v = zeros.(size.(p)) # Initialize
+        log!(M, v, p, q_)
+        return v
+    else
+        return Nothing
```

Comment:
Here the same as for Segre counts – returning nothing is maybe too type instable.